src/openeo_test_suite/lib/process_runner/base.py

```diff
@@ -0,0 +1,39 @@
+from typing import List, Dict, Any
+
+class ProcessTestRunner:
+
+  # Returns a list of all processes supported by the backend
+  # Follows definition of the same function in the openEO Python client
+  def list_processes(self) -> List[Dict]:
```

Comment:
I see you are using 2 spaces indentation. While valid, this is highly unusual in the Python world. I'd would recommend 4 spaces

Reply:
in general I recommend setting up the pre-commit hooks as discussed at https://github.com/Open-EO/openeo-test-suite#development-and-contributions

Reply:
I can surely update the indentation, 2 spaces is just my system default.

I can't run pre-commit unfortunately:

---

mm@Matthias-Workstation:/mnt/c/Dev/openeo-test-suite$ pre-commit run --all-files
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/python3', '-mvirtualenv', '/home/mm/.cache/pre-commit/repocmxw0_tg/py_env-python3.11', '-p', 'python3.11')
return code: 1
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.11'
stderr: (none)
Check the log at /home/mm/.cache/pre-commit/pre-commit.log
mm@Matthias-Workstation:/mnt/c/Dev/openeo-test-suite$

Reply:
The .pre-commit-config.yaml says language-version 3.11. As the pyproject says 3.9, I've updated the pre-commit accordingly, now it's working.

Reply:
Ok good point. I recently read it is even recommended to no specify a particular version  in  .pre-commit-config.yaml. as the defaults will  do the right thing anyway

Reply:
> 2 spaces is just my system default.

for python, or just the language agnostic default?